pgsql/pc_inout.c

```diff
@@ -313,6 +314,50 @@ Datum pcpatch_bytea_envelope(PG_FUNCTION_ARGS)
 	PG_RETURN_BYTEA_P(wkb);
 }
 
+PG_FUNCTION_INFO_V1(pcpatch_bounding_diagonal_as_bytea);
+Datum pcpatch_bounding_diagonal_as_bytea(PG_FUNCTION_ARGS)
+{
+	static const size_t stats_size_guess = 400;
+	SERIALIZED_PATCH *serpatch;
+	PCSCHEMA *schema;
+	PCSTATS *stats;
+	uint8 *bytes;
+	size_t bytes_size;
+	bytea *wkb;
+	size_t wkb_size;
+
+	serpatch = PG_GETHEADERX_SERPATCH_P(0, stats_size_guess);
+	schema = pc_schema_from_pcid(serpatch->pcid, fcinfo);
+
+	if ( schema->z_position > -1 || schema->m_position > -1 )
+	{
+		if ( stats_size_guess < pc_stats_size(schema) )
+			serpatch = PG_GETHEADERX_SERPATCH_P(0, pc_stats_size(schema));
+
+		stats = pc_patch_stats_deserialize(schema, serpatch->data);
+		if ( !stats )
+			PG_RETURN_NULL();
+
+		bytes = pc_bounding_diagonal_wkb_from_stats(stats, &bytes_size);
+
+		pc_stats_free(stats);
+	}
+	else
+	{
+		bytes = pc_bounding_diagonal_wkb_from_bounds(
+			&serpatch->bounds, schema, &bytes_size);
+	}
+
+	wkb_size = VARHDRSZ + bytes_size;
+	wkb = palloc(wkb_size);
+	memcpy(VARDATA(wkb), bytes, bytes_size);
+	SET_VARSIZE(wkb, wkb_size);
+
+	pfree(bytes);
```

Comment:
My remark is that `bytes` is allocated with `pcalloc`, so I would expect freeing it with `pcfree`. I do not know if it matters, though.